kernel: bump 5.4 to 5.4.62

graysky2 opened the pull request:
Manual merged:
```
 hack-5.4
  230-openwrt_lzma_options.patch  (with Adrian Schmutzler's help)
 bcm27xx
  950-0283-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch
 x86
  011-tune_lzma_options.patch (with Adrian Schmutzler's help)
```
Remove upstreamed patches in collaboration with Ansuel Smith
 ```ipq806x
  093-1-v5.8-ipq806x-PCI-qcom-Add-missing-ipq806x-clocks-in-PCIe-driver.patch
  093-2-v5.8-ipq806x-PCI-qcom-Change-duplicate-PCI-reset-to-phy-reset.patch
  093-3-v5.8-ipq806x-PCI-qcom-Add-missing-reset-for-ipq806x.patch
```
All other modifications made by `update_kernel.sh`

Build system: x86_64
Run-tested: ipq806x (R7800)

No dmesg regressions, everything functional

Signed-off-by: John Audia <graysky@archlinux.us>

graysky2 commented:
@Ansuel

adschm commented on target/linux/generic/hack-5.4/230-openwrt_lzma_options.patch:
```diff
@@ -27,11 +27,11 @@ Signed-off-by: Imre Kaloz <kaloz@openwrt.org>
  quiet_cmd_lzma = LZMA    $@
--      cmd_lzma = { cat $(real-prereqs) | lzma -9; $(size_append); } > $@
+-      cmd_lzma = { cat $(real-prereqs) | $(LZMA) -9; $(size_append); } > $@
 +      cmd_lzma = { cat $(real-prereqs) | lzma e -d20 -lc1 -lp2 -pb2 -eos -si -so; $(size_append); } > $@
```
Should we convert the "new" line to $(LZMA) here as well?
(This will require a manual merge for x86/011-tune_lzma_options.patch as well).

graysky2 replied:
@adschm - Would you mind making the proposed modification?

graysky2 replied:
I updated the PR based on your modifications to `target/linux/generic/hack-5.4/230-openwrt_lzma_options.patch` and `target/linux/x86/patches-5.4/011-tune_lzma_options.patch`

graysky2 replied:
New commit builds (x86_64) and runs (ipq806x).

Ansuel commented:
Can confirm that 
```
 093-1-v5.8-ipq806x-PCI-qcom-Add-missing-ipq806x-clocks-in-PCIe-driver.patch
 093-2-v5.8-ipq806x-PCI-qcom-Change-duplicate-PCI-reset-to-phy-reset.patch
 093-3-v5.8-ipq806x-PCI-qcom-Add-missing-reset-for-ipq806x.patch
```

Has been backported and can be dropped.

adschm commented:
Just checked with the recently merged layerscape patch, but that one only has cosmetic diff after refresh, so no need to do it again for this bump.

nicklowe commented:
For information, there is an important ath10k patch included in 5.4.62 that is applicable for OpenWRT users who are not using a CT driver with Wave 2 QCA 802.11ac generation hardware, it is already merged for CT.

It resolves a PCIe hang seen due to the DMA burst size value meaning different things between Wave 1 and Wave 2 hardware. The same change is also merged for 4.4.235, 4.9.235, 4.14.196 and 4.19.143

The issue resolved is that a value of 0 for the DMA burst size has a different meaning and instruction depending on the generation. Wave 1 and Wave 2 QCA 802.11ac generation hardware use this differently.

graysky2 commented:
@nicklowe - To be clear, you are just calling out a specific fix rather than calling out some action that needs to taken to enable, yes?

adschm commented:
> For information, there is an important ath10k patch included in 5.4.62 that is applicable for OpenWRT users who are not using a CT driver with Wave 2 QCA 802.11ac generation hardware, it is already merged for CT.
> 
> It resolves a PCIe hang seen due to the DMA burst size value meaning different things between Wave 1 and Wave 2 hardware. The same change is also merged for 4.4.235, 4.9.235, 4.14.196 and 4.19.143
> 
> The issue resolved is that a value of 0 for the DMA burst size has a different meaning and instruction depending on the generation.

Is this actually relevant here, as we have mac80211 separate?

anomeome commented:
compile / test : mvebu / mamba, rango

lucize commented:
this patch seems to fix the issue with 10mbps upload speed on LAN ports with a smart switch attached on wrt3200acm
I'll see if I get the quick crash for the wlan0 (5g)that appeared I don't know when, maybe my unit is faulty as I see no one complains about it in forum

```
ieee80211 phy0: cmd 0x801d=MEMAddrAccess timed out
ieee80211 phy0: return code: 0x001d
ieee80211 phy0: timeout: 0x001d
ieee80211 phy0: MACREG_REG_INT_CODE: 0x0000 
hostapd: nl80211: nl80211_recv_beacons->nl_recvmsgs failed: -5
```

graysky2 committed:
kernel: bump 5.4 to 5.4.63

Manual merged:
 hack-5.4
  230-openwrt_lzma_options.patch (with Adrian Schmutzler's help)
 bcm27xx
  950-0283-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch
 x86
  patches-5.4
	 011-tune_lzma_options.patch (with Adrian Schmutzler's help)

Remove upstreamed patches in collaboration with Ansuel Smith:
 ipq806x
  093-1-v5.8-ipq806x-PCI-qcom-Add-missing-ipq806x-clocks-in-PCIe-driver.patch
  093-2-v5.8-ipq806x-PCI-qcom-Change-duplicate-PCI-reset-to-phy-reset.patch
  093-3-v5.8-ipq806x-PCI-qcom-Add-missing-reset-for-ipq806x.patch

All other modifications made by update_kernel.sh

Build system: x86_64
Run-tested: ipq806x (R7800)

No dmesg regressions, everything functional

Signed-off-by: John Audia <graysky@archlinux.us>

graysky2 commented:
I made made a mistake when updating to 5.4.63 so I will close this PR and send a fresh on for it since it supersedes this.
